### ISSUE TITLE
prevent re rendering of server rendered component on the client with same data

### ISSUE THREAD

**Issue by laterne** [dekujs/deku#190]
I think react calculates a checksum of the server rendered html string and adds this as a data property to the root node. On the client it can compare the checksum to prevent a re render and therefore the flickering that occurs if the components get re rendered with the same data. During mounting it only attaches the event handlers.
Would it architecturally be possible to add this functionality to deku?

**Comment by anthonyshort** [comment]
Yeah it's definitely possible. At one point I had it in there. It creates a hash of the virtual tree. The part we've never had is the ability to assign pre-rendered virtual nodes to components, but it's definitely possible.
